apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +460,35 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (isSingleContactEnabled()) {
+			let { contactId } = visitor;
+			if (contactId) {
+				const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
+					projection: { contactId: 1 },
+				});
+				contactId = visitorContact?.contactId;
+			}
+
+			const contact = await LivechatContacts.findOne({ _id: contactId });
+
+			if (contact) {
+				const channel = contact.channels?.find((channel) => channel.name === roomInfo.source?.type && channel.visitorId === visitor._id);
+
+				if (!channel) {
+					Livechat.logger.debug(`Adding channel for contact ${contact._id}`);
+
+					await LivechatContacts.addChannel(contact._id, {
+						name: roomInfo.source?.label || roomInfo.source?.type.toString() || 'other',
+						visitorId: visitor._id,
+						blocked: false,
+						verified: false,
+						details: roomInfo.source,
+						channelIdentifier: roomInfo.source?.destination,
```

Comment:
In this case, when a `destination` is on roomInfo, you'll have the same property twice, one inside `details.destination` and one on `channelIdentifier`. Maybe we don't need that.